node/src/protocol/mod.rs

```diff
@@ -246,42 +234,61 @@ impl MpcSignProtocol {
                 }
             }
 
-            // Establish the participants for this current iteration of the protocol loop. This will
-            // set which participants are currently active in the protocol and determines who will be
-            // receiving messages.
-            self.ctx.mesh.establish_participants(&contract_state).await;
-
             let state = {
                 let guard = self.state.read().await;
                 guard.clone()
             };
-            let state = match state.progress(&mut self).await {
+
+            let mut state = match state.progress(&mut self).await {
                 Ok(state) => state,
                 Err(err) => {
                     tracing::info!("protocol unable to progress: {err:?}");
-                    tokio::time::sleep(Duration::from_millis(1000)).await;
-                    continue;
-                }
-            };
-            let mut state = match state.advance(&mut self, contract_state).await {
-                Ok(state) => state,
-                Err(err) => {
-                    tracing::info!("protocol unable to advance: {err:?}");
-                    tokio::time::sleep(Duration::from_millis(1000)).await;
+                    tokio::time::sleep(Duration::from_millis(100)).await;
                     continue;
                 }
             };
+
             if let Err(err) = state.handle(&self, &mut queue).await {
                 tracing::info!("protocol unable to handle messages: {err:?}");
-                tokio::time::sleep(Duration::from_millis(1000)).await;
+                tokio::time::sleep(Duration::from_millis(100)).await;
                 continue;
             }
```

Comment:
we should move this to after `state.advance` just so the messages for state advancement are handled before we assign `state` to `self.state`